app/controllers/search_controller.rb

```diff
@@ -1,8 +1,19 @@
 class SearchController < ApplicationController
   def index
     search = params[:search].split(":").first
-    @repositories = policy_scope(Repository).search(search)
-    @teams = policy_scope(Team).search(search)
-    @namespaces = policy_scope(Namespace).search(search)
+    @search_query = params[:search]
+    @search_type = params[:type]
+
+    repositories = policy_scope(Repository).search(search)
+    @repositories = repositories.page(params[:page]).per(2)
+    @repositories_count = repositories.count()
+
+    teams = policy_scope(Team).search(search)
+    @teams = teams.page(params[:page]).per(2)
+    @teams_count = teams.count()
+
+    namespaces = policy_scope(Namespace).search(search)
+    @namespaces = namespaces.page(params[:page]).per(2)
+    @namespaces_count = namespaces.count()
```

Comment:
I think that more than 2 items would be better. I guess that you set this value just for testing purposes though :wink: